zerver/lib/streams.py

```diff
@@ -914,7 +936,13 @@ def add_filter_option(option: Q) -> None:
             # Don't bother going to the database with no valid sources
             return []
 
-    stream_dicts = [stream_to_dict(stream) for stream in streams]
+    recent_traffic = None
+    if not user_profile.realm.is_zephyr_mirror_realm:
+        # We do not need stream traffic data for streams in zephyr mirroring realm.
+        stream_ids = {stream.id for stream in streams}
+        recent_traffic = get_streams_traffic(stream_ids)
```

Comment:
Would it be cleaner to make `get_streams_traffic` just return None if `user_profile.realm.is_zephyr_mirror_realm` rather than having checks like this in multiple callers?